### PR TITLE
feat(compass-editor): add autocompleter for aggregation, use autocompleter in import pipeline modal COMPASS-6175

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch adds a new aggregation autocompleter and attaches it to the import-pipeline editor. This autocompleter will suggest pipeline stages if you are typing outside of the stage, and will fallback to the stage autocompleter if typing inside the stage:

|Outside Stage|Inside Stage|
|---|---|
|image|image|

### CODE REVIEW

File: packages/compass-editor/src/ace/query-autocompleter.test.ts [hunk 1]
```diff
@@ -1,273 +1,158 @@
-import type { Ace } from 'ace-builds';
```
Comment (gribnoysup) [inline packages/compass-editor/src/ace/query-autocompleter.test.ts:1]: Both these tests and stage-autocompleter tests were not changed, they are still testing the same things as before, I just moved them to use this `setupCompleter` test helper I added for aggregation-autocompleter tests

File: packages/compass-editor/src/ace/aggregation-autocompleter.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    // filter out everything from the sibling blocks that we run into. This can
+    // be error-prone, but good enough for our purposes
+    skip = Math.max(
+      0,
+      type(token, 'paren')
+        ? skip +
+            (type(token, 'rparen') ? 1 : -1) *
+              // ace groups tokens with the same type, so one rparen can be
+              // `{{{` so we change the stack accordingly
+              token.value.length
```
Comment (gribnoysup) [inline packages/compass-editor/src/ace/aggregation-autocompleter.ts:62]: If someone was wondering what was broken in my demo last Friday 😅

**Review by mabaasit — APPROVED** [review]
Looks good. 

nit: would it be better to hide stage operator completion when a user is not at stage operator level?

https://user-images.githubusercontent.com/1305718/196172187-925e1b0c-a0be-4f38-ab2a-a4fb9dbebea6.mov

nit: (small ux thing): can we show autocompleter on keyboard shortcut (like Ctrl+space)

**Comment by gribnoysup** [comment]
> nit: (small ux thing): can we show autocompleter on keyboard shortcut (like Ctrl+space)

I'll add it to the design sync, if we want to enable it, it should be consistent with other editors

**Review by Anemy — APPROVED** [review]

**Comment by gribnoysup** [comment]
> nit: would it be better to hide stage operator completion when a user is not at stage operator level?

If we want to fix it, then it's probably worth doing for other autocompleters that are suggesting autocompletes for broken code just based on token being of an identifier type. But also FWIW based on how simplistic ace autocompleters are I would not try to invest much time into covering corner cases

**Pull request merged**